usrsctplib/netinet/sctp_callout.c

```diff
@@ -77,60 +79,28 @@ uint32_t sctp_get_tick_count(void) {
  * SCTP_TIMERQ_LOCK protects:
  * - SCTP_BASE_INFO(callqueue)
  * - sctp_os_timer_next: next timer to check
- * - sctp_os_timer_current: current callout callback in progress
- * - sctp_os_timer_current_tid: current callout thread id in progress
- * - sctp_os_timer_waiting: some thread is waiting for callout to complete
- * - sctp_os_timer_wait_ctr: incremented every time a thread wants to wait
- *                           for a callout to complete.
  */
 static sctp_os_timer_t *sctp_os_timer_next = NULL;
-static sctp_os_timer_t *sctp_os_timer_current = NULL;
```

Comment:
Note that this and a lot of stuff below is just reverting the problematic commits that cause the deadlock. Again, look at individual commits for clearer diff.